### ISSUE TITLE
ovs setup failure due to undefined symbol

### ISSUE THREAD

**Issue by winglq** [openvstorage/framework#1184]
After I get volumedriver-no-dedup-server from unstable repo. I could successfully install the openvstorage, but when I ran ovs setup it failed with the following tb:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "ovs/lib/setup.py", line 39, in <module>
    from ovs.extensions.storageserver.storagedriver import StorageDriverConfiguration
  File "ovs/extensions/storageserver/storagedriver.py", line 28, in <module>
    from volumedriver.storagerouter import storagerouterclient
ImportError: /usr/lib/python2.7/site-packages/volumedriver/storagerouter/storagerouterclient.so: undefined symbol: _ZNK6google8protobuf7Message11GetTypeNameB5cxx11Ev

**Comment by winglq** [comment]
BTW, it's running on Centos7.

**Comment by wimpers** [comment]
@pploegaert can you reproduce ?

**Comment by pploegaert** [comment]
Volumedriver installation is indeed broken - as we no longer provide support for CentOS - closed
with --skip-broken is does continue however ...
```
...
--> Processing Dependency: liburcu-bp.so.1()(64bit) for package: volumedriver-base-6.0.0.0-1.x86_64
---> Package volumedriver-base.x86_64 0:6.0.0.0-1 will be installed
--> Processing Dependency: liburcu-cds.so.1()(64bit) for package: volumedriver-base-6.0.0.0-1.x86_64
--> Processing Dependency: liburcu-bp.so.1()(64bit) for package: volumedriver-base-6.0.0.0-1.x86_64
--> Finished Dependency Resolution
Error: Package: volumedriver-server-6.0.0.0-1.x86_64 (openvstorage)
           Requires: liburcu-cds.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-cds.so.1()(64bit)
           Available: userspace-rcu-0.8.5-1.fc22.i686 (fc22)
               Not found
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-cds.so.4()(64bit)
Error: Package: 1:librbd1-0.94.5-1.el7.x86_64 (base)
           Requires: libboost_thread-mt.so.1.53.0()(64bit)
           Available: boost-thread-1.53.0-26.el7.x86_64 (base)
               libboost_thread-mt.so.1.53.0()(64bit)
           Available: boost-thread-1.57.0-6.el7.centos.x86_64 (openvstorage)
               Not found
           Available: boost-thread-1.57.0-6.fc22.i686 (fc22)
               Not found
Error: Package: volumedriver-base-6.0.0.0-1.x86_64 (openvstorage)
           Requires: liburcu-bp.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-bp.so.1()(64bit)
           Available: userspace-rcu-0.8.5-1.fc22.i686 (fc22)
               Not found
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-bp.so.4()(64bit)
Error: Package: 1:librbd1-0.94.5-1.el7.x86_64 (base)
           Requires: libboost_system-mt.so.1.53.0()(64bit)
           Available: boost-system-1.53.0-26.el7.x86_64 (base)
               libboost_system-mt.so.1.53.0()(64bit)
           Available: boost-system-1.57.0-6.el7.centos.x86_64 (openvstorage)
               Not found
           Available: boost-system-1.57.0-6.fc22.i686 (fc22)
               Not found
Error: Package: 1:librados2-0.94.5-1.el7.x86_64 (base)
           Requires: libboost_system-mt.so.1.53.0()(64bit)
           Available: boost-system-1.53.0-26.el7.x86_64 (base)
               libboost_system-mt.so.1.53.0()(64bit)
           Available: boost-system-1.57.0-6.el7.centos.x86_64 (openvstorage)
               Not found
           Available: boost-system-1.57.0-6.fc22.i686 (fc22)
               Not found
Error: Package: 1:librados2-0.94.5-1.el7.x86_64 (base)
           Requires: libboost_thread-mt.so.1.53.0()(64bit)
           Available: boost-thread-1.53.0-26.el7.x86_64 (base)
               libboost_thread-mt.so.1.53.0()(64bit)
           Available: boost-thread-1.57.0-6.el7.centos.x86_64 (openvstorage)
               Not found
           Available: boost-thread-1.57.0-6.fc22.i686 (fc22)
               Not found
Error: Package: volumedriver-base-6.0.0.0-1.x86_64 (openvstorage)
           Requires: liburcu-cds.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-cds.so.1()(64bit)
           Available: userspace-rcu-0.8.5-1.fc22.i686 (fc22)
               Not found
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-cds.so.4()(64bit)
Error: Package: volumedriver-server-6.0.0.0-1.x86_64 (openvstorage)
           Requires: liburcu-bp.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-bp.so.1()(64bit)
           Available: userspace-rcu-0.8.5-1.fc22.i686 (fc22)
               Not found
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-bp.so.4()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Installing a recent volumedriver pkg results in dependency problems - to be investigated if still necessary:

```
--> Finished Dependency Resolution
Error: Package: volumedriver-no-dedup-base-6.8.1.0-1.x86_64 (/volumedriver-no-dedup-base_6.8.1.0-1)
           Requires: liburcu-bp.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-bp.so.1()(64bit)
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-bp.so.4()(64bit)
Error: Package: volumedriver-no-dedup-base-6.8.1.0-1.x86_64 (/volumedriver-no-dedup-base_6.8.1.0-1)
           Requires: liburcu-cds.so.1()(64bit)
           Available: userspace-rcu-0.7.16-1.el7.x86_64 (epel)
               liburcu-cds.so.1()(64bit)
           Installing: userspace-rcu-0.9.3-1.el7.x86_64 (EfficiOS-RHEL7-x86-64)
              ~liburcu-cds.so.4()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

**Comment by pploegaert** [comment]
@saelbrec: pls comment if still necessary - thx

**Comment by wimpers** [comment]
It seems CentOS is currently broken. I will create a new ticket to gather all tasks around CentOS.

**Comment by run283092610** [comment]
i also running on centos7, when I ran ovs setup it failed 

[root@localhost site-packages]# ovs setup
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "ovs/lib/nodeinstallation.py", line 38, in <module>
    from ovs.lib.nodetype import NodeTypeController
  File "ovs/lib/nodetype.py", line 36, in <module>
    from ovs.extensions.storageserver.storagedriver import StorageDriverConfiguration
  File "ovs/extensions/storageserver/storagedriver.py", line 26, in <module>
    from volumedriver.storagerouter import storagerouterclient
ImportError: /usr/lib/python2.7/site-packages/volumedriver/storagerouter/storagerouterclient.so: undefined symbol: _ZN6snappy10UncompressEPKcmPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
